packages/common/src/store/wallet/utils.ts

```diff
@@ -48,37 +49,35 @@ export const badgeTiers: BadgeTierInfo[] = [
 
 export const getVerifiedForUser = (
   state: CommonState,
-  { userId }: { userId: ID }
+  { userId }: { userId?: ID }
 ) => {
   const user = getUser(state, { id: userId })
   return !!user?.is_verified
 }
 
-export const getWeiBalanceForUser = (
-  state: CommonState,
-  { userId }: { userId: ID }
-) => {
-  const accountUser = getAccountUser(state)
-  const user = getUser(state, { id: userId })
-
-  if (accountUser?.user_id === userId && state.wallet.totalBalance) {
-    return state.wallet.totalBalance
-  }
-  if (!user) return '0' as StringWei
-  return getUserBalance(user)
+/**
+ * Hook to get the tier and verification status for a user
+ * @param userId Optional user ID to check. If not provided, uses the current user
+ * @returns Object containing tier, isVerified, and tierNumber
+ */
+export const useTierAndVerifiedForUser = (userId?: ID | null) => {
+  const { data: currentUser } = useCurrentAccountUser()
```

Comment:
fine for now but might be nice to just select what's needed and getUserBalance to update the minimum user it needs